version: "3.8"
services:
  # bot:
  #   build:
  #     context: .
  #     dockerfile: bot/Dockerfile
  #   ports:
  #     - "8081:8080"
  #   volumes:
  #     - .:/app
  #   command: ["python", "bot/__main__.py"]

  db:
    image: postgres:15
    environment:
      POSTGRES_USER: ${DB_USER}
      POSTGRES_PASSWORD: ${DB_PASSWORD}
      POSTGRES_DB: ${DB_NAME}
    volumes:
      - pgdata:/var/lib/postgresql/data
    ports:
      - "5432:5432"
    healthcheck:
      test: ["CMD-SHELL", "pg_isready -U ${DB_USER}"]
      interval: 5s
      timeout: 5s
      retries: 5

  app:
    build: 
      context: ./backend
      dockerfile: Dockerfile
    ports:
      - "8080:8080"
    depends_on:
      db:
        condition: service_healthy
    environment:
      APP_PORT: ${APP_PORT}
      LOG_LEVEL: ${LOG_LEVEL}
      API_TOKEN: ${API_TOKEN}
      OUTLINE_API_URL: ${OUTLINE_API_URL}
      OUTLINE_API_KEY: ${OUTLINE_API_KEY}
      DB_HOST: ${DB_HOST}
      DB_PORT: ${DB_PORT}
      DB_USER: ${DB_USER}
      DB_PASSWORD: ${DB_PASSWORD}
      DB_NAME: ${DB_NAME}
    # tty: true

volumes:
  pgdata:
